src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets

```diff
@@ -195,35 +192,27 @@ Copyright (c) .NET Foundation. All rights reserved.
          is a NativeAOT app, value of SelfContained doesn't matter. -->
     <NETSdkError Condition="'$(RunILLink)' != 'false' And '$(SelfContained)' != 'true'" ResourceName="ILLinkNotSupportedError" />
 
+    <Warning Condition="'$(SuppressILLinkExplicitPackageReferenceWarning)' != 'true' And
+                        '%(PackageReference.Identity)' == 'Microsoft.NET.ILLink.Tasks' And '%(PackageReference.IsImplicitlyDefined)' != 'true'"
+             Text="Delete explicit 'Microsoft.NET.ILLink.Tasks' package reference in your project file. Explicit 'Microsoft.NET.ILLink.Tasks' package reference can run into version errors." />
```

Comment:
```suggestion
             Text="Delete explicit 'Microsoft.NET.ILLink.Tasks' package reference in your project file to avoid versioning problems." />
```